server/server.go

```diff
@@ -685,6 +685,9 @@ func New(opts *Options) *Server {
 // The provided Options type should not be re-used afterwards.
 // Either use Options.Clone() to pass a copy, or make a new one.
 func NewServer(opts *Options) (*Server, error) {
+	if opts.ConfigFile != _EMPTY_ {
+		opts.ProcessConfigFile(opts.ConfigFile)
+	}
```

Comment:
it would be surprising if NewServer() changes my passed options like this without some opt-in.

It might be better to add a helper function like `NewServerFromConfig(file string)` or something like that?  Else maybe gate this behaviour on having passed a nil options with clear docs callout about that - but that doesn't seem so nice.

Reply:
The mentioned issue also uses some other options instead of just the config file.
Why not call `opts.ProcessConfigFile(opts.ConfigFile)` right before calling `NewServer` on the options when embedding? (that's the same thing the server already does when combining CLI flags, etc.)